client/state-db/src/lib.rs

```diff
@@ -663,14 +706,26 @@ fn choose_pruning_mode(
 #[cfg(test)]
 mod tests {
 	use crate::{
+		noncanonical::LAST_CANONICAL,
+		pruning::LAST_PRUNED,
 		test::{make_changeset, make_db, TestDb},
-		Constraints, Error, PruningMode, StateDb, StateDbError,
+		to_meta_key, CommitSet, Constraints, Error, IsPruned, PruningMode, StateDb, StateDbError,
 	};
+	use codec::Encode;
 	use sp_core::H256;
 	use std::io;
 
-	fn make_test_db(settings: PruningMode) -> (TestDb, StateDb<H256, H256>) {
+	fn make_test_db(settings: PruningMode) -> (TestDb, StateDb<H256, H256, TestDb>) {
 		let mut db = make_db(&[91, 921, 922, 93, 94]);
+
+		let mut commit = CommitSet::default();
+		commit
+			.meta
+			.inserted
+			.push((to_meta_key(LAST_CANONICAL, &()), (H256::from_low_u64_be(0), 0u64).encode()));
+		commit.meta.inserted.push((to_meta_key(LAST_PRUNED, &()), 0u64.encode()));
+		db.commit(&commit);
```

Comment:
This should not be required really. When starting with an empty database there's not last pruned and last canonical block until they are written. What's the reason for adding this?

Reply:
It is because the test insert blocks start from block number 1, which should be 0 if it is an empty database, I added these meta keys to avoid changing the original test a lot, the case `start from an empty database` is covered by `init_db_backed_queue` though.